src/Components/Applications/Form/FormApplication.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import styles from './form.module.css';
+
+const FormApplication = () => {
+  const [position, setPositionName] = useState('');
+  const [company, setCompany] = useState('');
+  const [postulant, setPostulant] = useState('');
+  const [applicationState, setAppState] = useState("");
+
+   const onChangePosition = (event) => {
+    setPositionName(event.target.value);
+  };
+  const onChangeCompany = (event) => {
+    setCompany(event.target.value);
+  };
+  const onChangePostulant = (event) => {
+    setPostulant(event.target.value);
+  };
+  const onChangeAppState = (event) => {
+    setAppState(event.target.value);
+    console.log(event.target.value);
+  };
+
+  const params = new URLSearchParams(window.location.search);
+  const AppId = params.get('id');
+
+  useEffect(()=> {
+    if(AppId){
+      fetch(`${process.env.REACT_APP_API}/api/applications/id/${AppId}`)
+        .then((response) => {
+          if (response.status !== 200) {
+            return response.json().then(({ ErrMessage }) => {
+              throw new Error(ErrMessage);
+            });
+          }
+          return response.json();
+        })
+        .then((response) => {
+          setPositionName(response.positionId);
+          setCompany(response.companyId);
+          setPostulant(response.postulantId ? response.postulantId :"No id"); // Bad DB. Applications with no postulantID exist
+          setAppState(response.applicationState);
+        });
+    }
+  }, []);
+  console.log (position, company,postulant, applicationState );
+  const onSubmit = (event) => {
+    event.preventDefault();
+    let url;
+    const options = {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        positionId: position,
+        companyId: company,
+        postulantId: postulant,
+        applicationState: applicationState
+      })
+    };
+
+    if (AppId === null) {
+      options.method = 'POST';
+      url = `${process.env.REACT_APP_API}/api/applications/add`;
```

Comment:
If you check on the deployed app in [heroku](https://basd-2021-aq-ms-18-appl-fikann.herokuapp.com/applications) , the requests to the server are failing cause `/api` is duplicated. It is probably that `/api` already included in the 
`REACT_APP_API` environment variable. You should remove it from here and add it in your `.env` file

![image](https://user-images.githubusercontent.com/8579539/144464025-a62bdb30-81bc-4f9a-80b9-6d51f7836060.png)


```suggestion
      fetch(`${process.env.REACT_APP_API}/clients/id/${clientId}`)
```